src/edu/csus/ecs/pc2/core/report/ResultsCompareReport.java

```diff
@@ -0,0 +1,369 @@
+            AwardKey awardsKey = new FileComparisonUtilities.AwardKey();
+            ScoreboardJSONKey scoreboardKey = new FileComparisonUtilities.ScoreboardJSONKey();
+
+            FileComparison resultsCompare = FileComparisonUtilities.createTSVFileComparison(ResultsFile.RESULTS_FILENAME, sourceDir, targetDir, resultTSVKey);
+            FileComparison awardsFileCompare = FileComparisonUtilities.createAwardJSONFileComparison(Constants.AWARDS_JSON_FILENAME, sourceDir, targetDir, awardsKey);
+            FileComparison scoreboardJsonCompare = FileComparisonUtilities.createScoreboardJSONFileComparison(Constants.SCOREBOARD_JSON_FILENAME, sourceDir, targetDir, scoreboardKey);
+
+            long totalDifferences = //
+                    resultsCompare.getNumberDifferences() + //
+                    awardsFileCompare.getNumberDifferences() + //
+                    scoreboardJsonCompare.getNumberDifferences();
+
+            if (totalDifferences == 0) {
+                outList.add("Summary:   All PASS - no differences found");
+                outList.add("");
+                
+                outList.add(String.format("%3d", scoreboardJsonCompare.getNumberRows())+ " rows "+Constants.SCOREBOARD_JSON_FILENAME);
+                outList.add(String.format("%3d", awardsFileCompare.getNumberRows())+ " rows "+Constants.AWARDS_JSON_FILENAME);
+                outList.add(String.format("%3d", resultsCompare.getNumberRows())+ " rows "+ResultsFile.RESULTS_FILENAME);
+            } else {
+                outList.add("Summary:   FAIL - there are " + totalDifferences + "  differences found");
+                outList.add("");
+
+                outList.add(String.format("%3d", scoreboardJsonCompare.getNumberDifferences())+ " differences "+Constants.SCOREBOARD_JSON_FILENAME);
+                outList.add(String.format("%3d", awardsFileCompare.getNumberDifferences())+ " differences "+Constants.AWARDS_JSON_FILENAME);
+                outList.add(String.format("%3d", resultsCompare.getNumberDifferences())+ " differences "+ResultsFile.RESULTS_FILENAME);
+            }
+
+
+            String[] infoLines = { //
+                    "", //
+                    "Primary CCS results dir: " + getPrimaryCCSResultsDir(), //
+                    "PC2 results dir        : " + getPc2ResultsDir(), //
+                    "", //
+            };
+            outList.addAll(Arrays.asList(infoLines));
+
+            if (fullDetails) {
+
+                List<String> lines = createFileComparisonReport(ResultsFile.RESULTS_FILENAME, resultsCompare);
+                outList.addAll(lines);
+                lines = createFileComparisonReport(Constants.AWARDS_JSON_FILENAME, awardsFileCompare);
+                outList.addAll(lines);
+                lines = createFileComparisonReport(Constants.SCOREBOARD_JSON_FILENAME, scoreboardJsonCompare);
+                outList.addAll(lines);
+            }
+        }
```

Comment:
This bracket is the end of the `if (new File(targetDir).isDirectory()) {` condition above (and which as indicated above should be `if (new File(targetDir).isDirectory()  &&  new File(sourceDir).isDirectory()) {`.   What happens if the `targetDir` (or the `sourceDir`) is NOT a directory?  There should be an `else` here that reports this error condition -- at least by logging it, and probably also by noting it in the created report.  Otherwise the user will see a failed comparison but have little idea why it failed.

I consider this a SHOW STOPPER that needs to be fixed before I would Approve the PR.

Reply:
If target or source are missing the report should state that either or both directories do not exist.

Reply:
> If target or source are missing the report should state that either or both directories do not exist.

Agreed.  Eagerly awaiting an updated PR with such a fix (along with inclusion of logging the problem).